### PR TITLE
Clear geography data

### PR DESCRIPTION
**Pull request by PaulWessel**
**Description of proposed changes**

New target added to clear: _geography_[=_name_], where name can be either gshhg or dcw.  Default deletes both.

### CODE REVIEW

File: src/clear.c [hunk 1]
```diff
@@ -20,7 +20,7 @@
  * Version:	6 API
  *
  * Brief synopsis: gmt clear cleans up by removing files or dirs.
- *	gmt clear [all | cache | data[=<planet>] | sessions | settings ]
+ *	gmt clear [all | cache | data[=<planet>] | geography | sessions | settings ]
```
Comment (seisman) [inline src/clear.c:23]:
```suggestion
 *	gmt clear [all | cache | data[=<planet>] | geography[=<name>] | sessions | settings ]
```

File: src/clear.c [hunk 2]
```diff
@@ -29,22 +29,24 @@
...
+#define THIS_MODULE_PURPOSE	"Delete current default settings, or the cache, data, geography or sessions directories"
 #define THIS_MODULE_KEYS	""
 #define THIS_MODULE_NEEDS	""
 #define THIS_MODULE_OPTIONS	"V"
 
 static int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s all|cache|data[=<planet>]|sessions|settings [%s]\n\n", name, GMT_V_OPT);
+	GMT_Message (API, GMT_TIME_NONE, "usage: %s all|cache|data[=<planet>]|geography[=<name>|sessions|settings [%s]\n\n", name, GMT_V_OPT);
```
Comment (seisman) [inline src/clear.c:40]:
```suggestion
	GMT_Message (API, GMT_TIME_NONE, "usage: %s all|cache|data[=<planet>]|geography[=<name>]|sessions|settings [%s]\n\n", name, GMT_V_OPT);
```

File: src/clear.c [hunk 3]
```diff
@@ -174,6 +176,46 @@ static int clear_sessions (struct GMTAPI_CTRL *API) {
...
+		return GMT_NOERROR;
+	}
+
+	d = 0;
+	while (dir[d]) {	/* Look through geography subdirectories */
+		if (data && strcmp (dir[d], data)) {	/* Not a dataset that we want to delete */
+			d++;
+			continue;
+		}
+		sprintf (current_dir, "%s/%s", geography_dir, dir[d]);	/* E.g., ~/.gmt/server/gshhg */
```
Comment (seisman) [inline src/clear.c:201]:
```suggestion
		sprintf (current_dir, "%s/%s", geography_dir, dir[d]);	/* E.g., ~/.gmt/geography/gshhg */
```

**Comment by PaulWessel** [comment]
Sorry, tested the geography part but then added the =dcw|gshhg without testing.  Think it should be OK now.

**Pull request merged**